aip/general/0236.md

```diff
@@ -0,0 +1,315 @@
+  - Remains declarative friendly as only an output only field is modified.
+- If startPreview is called specifically on a resource which is already being
+  previewed, then it **must** return an error message.
+- When startPreview is called, the system **must** set values for the
+  `log_prefix`, and `start_time` in `PreviewMetadata` and set the state to ACTIVE.
+- If startPreview is called on an experiment with an empty set of rules, the
+  deletion of that resource **must** be previewed.
+- startPreview is not idempotent and using it twice on the same resource
+  **must** return an error.
+- Etags **must** be used to guarantee resource freshness.
+
+#### stopPreview
+
+```proto
+rpc StopPreviewPolicyExperiment(StopPreviewPolicyExperimentRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    post: "/v1/{name=policies/*/experiments/*}:stopPreview"
+    body: "*"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "PolicyExperiment"
+    metadata_type: "StopPreviewPolicyExperimentMetadata"
+  };
+}
+
+message StopPreviewPolicyExperimentRequest {
+  string name = 1;
+  string etag = 2;
+}
+
+message StopPreviewPolicyExperimentMetadata {}
+```
+- Admin flow API to stop previewing a resource **must** be implemented.
+  - Remains declarative friendly as only an output only field is modified.
+- If stopPreview is called specifically on a resource which is not actively
+  being previewed, then it **must** return an error message.
+- When the API is called, the `state` **must** be set to INACTIVE and
+  `stop_time` **must** be set to the current time.
+- stopPreview is not idempotent, calling stop twice on the same resource
+  **must** return an error.
+- Etags **must** be used to guarantee resource freshness.
+
+#### commit
+
+A new API called commit **may** be implemented to promote the experimental
+configuration of a resource to the effective version and perform necessary
+cleanup. It will copy the fields-under-test from the experiment into the live
+policy and then delete the experiment.
+
+This is a convenience API which is not declarative friendly. The alternative
+supported for declarative clients is to manually copy fields from an experiment
+onto the live Policy then delete the experiment.
+
+```proto
+rpc CommitPolicyExperiment(CommitPolicyExperimentRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    post: "/v1/{name=policies/*/experiments/*}:commit"
+    body: "*"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "google.protobuf.Empty"
+    metadata_type: "CommitPolicyExperimentMetadata"
+  };
+}
+
+message CommitPolicyExperimentRequest {
+  string name = 1;
+  string etag = 2;
+  string parent_etag = 3; // optional.
+}
+
+message CommitPolicyExperimentMetadata {}
+```
+- Commit **must** atomically copy the fields-under-test from the experiment into
+  the live Policy then deletes the experiment.
```

Comment:
Subject verb confusion.

"...live Policy and then delete the experiment."

is correct.